codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -79,19 +119,47 @@ public T get(T name, T defaultValue) {
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public T getAndRemove(T name) {
```

Comment:
This method implementation seems odd. We always have to traverse the whole bucket, why the two `for(;;)`. Assuming @trustin wrote this originally?